File: CS/ViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel;

namespace DataContextWpfSample {

    public class DataStoreViewModel {
        DataStoreModel dataStore;

        public DataStoreViewModel(DataStoreModel dataStore) {
            if (dataStore == null)
                throw new ArgumentNullException("dataStore");
            this.dataStore = dataStore;
        }
        public string ModelConnectionString { get { return dataStore.ConnectionString; } }
    }
}

File: VB/ViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.ComponentModel

Namespace DataContextWpfSample

	Public Class DataStoreViewModel
		Private dataStore As DataStoreModel

		Public Sub New(ByVal dataStore As DataStoreModel)
			If dataStore Is Nothing Then
				Throw New ArgumentNullException("dataStore")
			End If
			Me.dataStore = dataStore
		End Sub
		Public ReadOnly Property ModelConnectionString() As String
			Get
				Return dataStore.ConnectionString
			End Get
		End Property
	End Class
End Namespace
